### PR TITLE
Initial refactor of rainbowup

### PR DESCRIPTION
**Pull request by chefsale**

### CODE REVIEW

File: rainbowup/scripts/prepare.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
+CORE_SRC="~/.rainbowup/core"
```
Comment (ailisp) [inline rainbowup/scripts/prepare.sh:1]: Suggest adding #!/bin/bash to be more deterministic

File: rainbowup/scripts/prepare.sh [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+       BRIDGE_SRC="$2"
+       shift 2 ;;
+    *)
+       break ;;
+  esac
+done
+
+# Compile source of nearcore
+cd $CORE_SRC && cargo build --package neard --bin neard
+echo "Compiled source of nearcore"
```
Comment (ailisp) [inline rainbowup/scripts/prepare.sh:21]: I feel this and below echos good self documented, so these comments are not very necessary

File: environment/package.json [hunk 3]
```diff
@@ -7,13 +7,15 @@
...
     "@openzeppelin/cli": "^2.8.1",
+    "commander": "^5.1.0",
     "eth-object": "^1.0.3",
     "eth-util-lite": "^1.0.1",
     "ethereumjs-block": "^2.2.2",
     "ethereumjs-util": "^6.2.0",
     "ganache-cli": "^6.9.0",
     "merkle-patricia-tree": "^3.0.0",
     "nearlib": "^0.22.0",
+    "pm2": "^4.4.0",
```
Comment (ailisp) [inline environment/package.json:18]: Maybe better to be a devdependency

File: rainbowup/rainbowuplib/ethprover_tester_service.py [hunk 4]
```diff
@@ -3,52 +3,58 @@
...
+          'host',
+          '-v',
+          f'{client_contract_path}:{client_contract_path}',
+          '-v',
+          f'{prover_contract_path}:{prover_contract_path}',
+          *env_list,
+          self.args.rainbow_environment_image,
+          'start',
+          'test_ethprover',
+      ])
```
Comment (ailisp) [inline rainbowup/rainbowuplib/ethprover_tester_service.py:54]: This format looks nice! I suggest commit your yapf config file too

File: environment/package.json (hunk 3 above)
Comment (chefsale) [reply environment/package.json:18]: this is an actual dependency, not a devdependency

File: rainbowup/rainbowuplib/ethprover_tester_service.py (hunk 4 above)
Comment (chefsale) [reply rainbowup/rainbowuplib/ethprover_tester_service.py:54]: will do so.

File: rainbowup/scripts/prepare.sh (hunk 2 above)
Comment (chefsale) [reply rainbowup/scripts/prepare.sh:21]: done

File: rainbowup/scripts/prepare.sh [hunk 5]
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
```
Comment (nearmax) [inline rainbowup/scripts/prepare.sh:1]: This script does not work when location of nearcore and bridge is not in `~/.rainbowup`.

Comment (chefsale) [reply rainbowup/scripts/prepare.sh:1]: This was semi implemented.

**Pull request merged**